File: c++/write.cc
#include "write.h"
#include <iostream>

using namespace std;

void write::ejecutar(stack<int>& pila){
    int valor = pila.top();
    pila.pop();
    cout << valor << endl;
}
int write::modificar(int pc){
    return pc + 1;
}

std::string write::mostrar() {
    return " write";
}

File: java/write.java
import java.util.Stack;

class write extends Instruccion {
    @Override 
    public void ejecutar(Stack<Integer> pila){
        System.out.println(pila.pop()); // Desapilo el valor de la pila y lo escribo en la salida estándar
    }
    
    @Override 
    public int incrementarPc(int pc){
        return pc + 1;  // Incremento el contador de programa en 1
    }

    @Override
    public String mostrar(){
        return "    write";
    }
}
